jwst/assign_wcs/nirspec.py

```diff
@@ -748,8 +754,15 @@ def oteip_to_v23(reference_files):
         ote = f.tree['model'].copy()
     fore2ote_mapping = Identity(3, name='fore2ote_mapping')
     fore2ote_mapping.inverse = Mapping((0, 1, 2, 2))
-    # Convert the wavelength to microns
-    return fore2ote_mapping | (ote & (Identity(1)))# / Const1D(1e-6)))
+
+    # Create the transform to v2/v3/lambda.  The wavelength units up to this point are
+    # meters as required by the pipeline but the desired output wavelength units is microns
+    # as the expected output is microns.
+    # So we are going to Scale the data by multiplying the spatial units by 1 and
+    # the spectral units by 1e6 (meters -> microns)
+    lambda_meters_to_microns = (Scale(1) & Scale(1) & Scale(1e6))
+
+    return fore2ote_mapping | (ote & (Identity(1))) | lambda_meters_to_microns
```

Comment:
I made the change and re ran the tests and they worked. Same output as is shown above:
```
n [5]: w1(x[:3,:3], y[:3,:3])
Out[5]:
(array([[ 0.08277392,  0.08277426,  0.08277459],
        [ 0.08279353,  0.08279387,  0.0827942 ],
        [ 0.08281315,  0.08281348,  0.08281381]]),
 array([[ 0.00772621,  0.00772659,  0.00772697],
        [ 0.00774837,  0.00774875,  0.00774912],
        [ 0.00777055,  0.00777092,  0.0077713 ]]),
 array([[ 0.94127035,  0.94151151,  0.94175268],
        [ 0.9412465 ,  0.94148766,  0.94172883],
        [ 0.94122279,  0.94146396,  0.94170513]]))
```